### PR TITLE
Fix issue #9 - Issue with Bootstrap Modals

### PR DESCRIPTION
**Pull request by lydongray**
Fixes issue #9 with Bootstrap Modals by allowing user to specify where to place the scrollbar. Also addresses issue with resize event no longer firing when using 'fl-scrolls-body'.

### CODE REVIEW

File: src/jquery.floatingscroll.js [hunk 1]
```diff
@@ -134,8 +143,8 @@
                 cont = inst.cont,
                 pos = cont.getBoundingClientRect();
             inst.sbar.width($(cont).outerWidth());
-            if (!inst.scrollBody) {
-                inst.sbar.css("left", pos.left + "px");
+            if (inst.scrollContainer || !inst.scrollBody) {
```
Comment (Amphiluke) [inline src/jquery.floatingscroll.js]: Could you please explain why do we need to modify this condition? Let's suppose that the user has added both the `.fl-scrolls-container` element and the `.fl-scrolls-viewport`+`.fl-scrolls-body` couple. In this case we may get wrong positioning of the scroll bar as its left position is being relative to the `.fl-scrolls-viewport` element while `pos.left` is relative to the viewport.
Here is a demo: https://jsfiddle.net/oxpqpvh7/
And here is the same demo but with old variant of the condition: https://jsfiddle.net/7odr9yte/

**Review by Amphiluke — COMMENTED** [review]
One thing confuses me as I commented on the united diff page… Everything else looks OK

**Comment by lydongray** [comment]
Yes you're correct. I overlooked that 😕. Working on a slightly different approach.

**Comment by lydongray** [comment]
Hey,
so what I was trying to do initially was make a special case for the Bootstrap modal when in fact I should have been making a case for all situations.

In this commit, I have made three changes

1. The scrollbar will always position itself to the left position of the overflow panel.
2. The scrollbar will now position itself to the correct top position, depending on the scrollBody. If there is a scrollBody it will apply a "top" position relative to the scrollBody, otherwise default to the bottom of the window.
3. The CSS rule `.fl-scrolls-viewport .fl-scrolls {}` is now redundant because the scrollbar requires the position style to remain fixed. 

I have tested this in different situations and you can see my results here:
Basic example: http://embed.plnkr.co/IpksT5KPBiLADlVz8Wzs/
Simple modal example: http://embed.plnkr.co/uRJwCKL0jVh3iwiU7yT9/
Bootstrap modal example: http://embed.plnkr.co/DoR0IN12FbXPCVLRNwZL/
Your example: https://jsfiddle.net/lydongray/uncyde84/1/

**Comment by Amphiluke** [comment]
Hello,
First, thank you for the contribution! But let's go back to the source of the problem for a while…
You have provided some demos in your previous comment

> Basic example: http://embed.plnkr.co/IpksT5KPBiLADlVz8Wzs/
> Simple modal example: http://embed.plnkr.co/uRJwCKL0jVh3iwiU7yT9/
> Bootstrap modal example: http://embed.plnkr.co/DoR0IN12FbXPCVLRNwZL/

You are right, they seem to work as expected. But they also do so if we use the _current_ version of the plugin. All you need to do is just assign the indicating class names appropriately and probably customize styles a little bit.

Here are the same examples, but now they use the _current_ version of the plugin (without any modifications in the plugin sources).

Basic example: https://jsfiddle.net/5y80wu39/
Simple modal example: https://jsfiddle.net/drdbxscx/
Bootstrap modal example: https://jsfiddle.net/fpc18snk/

Seems to work as well. Just make sure that you assign the indicating class names to the appropriate elements.

With using the original Bootstrap sources I have also created a working demo on the base of an example setup provided by the [Bootstrap docs](http://getbootstrap.com/javascript/#live-demo). Here it is: 

https://jsfiddle.net/4qywso0z/

So do we really need to change something in the sources of the plugin to make it work with Bootstrap modals?

**UPD**
BTW, the example you provided in the issue #9 also starts working after assigning classes appropriately: https://jsfiddle.net/zswvqpsy/

**Comment by lydongray** [comment]
Thanks. Sorry for dragging this on for so long. 
I have a fairly complicated example that I was trying so hard to simplify (obviously without success). I probably should have started with this example in the first place. Anyway, what I'm trying to achieve is a scrollable (y axis) modal with scrollable (x axis) content. I'd applied some custom CSS to the modal to get the modal scrolling however was unable to get the scrollbar to behave.

https://jsfiddle.net/lydongray/zj6hLonw/1/

You can see how applying a 'bottom' position doesn't work when dealing with a 'transformed' element. So what I was trying to do was put the scrollbar into a different parent. This causes a number of side effects which I also tried to resolve.

https://jsfiddle.net/lydongray/r6bk5m87/1/

If there's a way to do this without any changes to the code, I'm all for that. The less changes and the lighter the code the better. However, I have been unable to solve this with the current version. Are you able to achieve this using the current version?

**Comment by Amphiluke** [comment]
Hello @lydongray,
I'm sorry for keeping the topic without an answer for such a long time, as I had no chance to return to it before.

First, regarding the example you showed in your last comment… Actually, yes, there is a way to make things work using the current version of the plugin. Well, it is not as straightforward as we probably want but nevertheless it seems to me that it's worth considering. So if the result you demonstrated in your last fiddle is really what you are trying to achieve then please check out this one:

https://jsfiddle.net/m96tm492/

Looks like it behaves just like yours but without any modifications in the plugin sources. I have wrapped the modal content with an additional block `.fl-scrolls-body` and have adjusted the styles slightly.

Anyway, the fact is that we can make up other cases when the plugin becomes less applicable. The question is how common those cases are. Perhaps in some cases even changing a widget's parent manually is not such a bad thing (of course if you know what are you doing)…

Now regarding your merge request. Sorry but I think that the suggested changes are not fully acceptable at the moment. 

* Firstly these changes break backward compatibility. Thus, the [“official” demo](http://amphiluke.github.io/floating-scroll/#demo-popup) appears to be broken when applying them.
* Second, [changing the top position](https://github.com/Amphiluke/floating-scroll/pull/10/files#diff-1a7171cfe1d3a5129fecc02eef4c6461R120) of the scroll bar each time the scroll event is triggered seems to me kind of an overkill which may potentially cause some performance issues… This way of handling the event provokes the Firefox Dev Tools to warn as in the screenshot below.
warning
A warning induced by the plugin is definitely not what we want to have in the production. Moreover I think that it's not very good semantically to place the code which adjusts scroll bar position into the function designed just to toggle scroll bar visibility.
* The third thing is an error in code style reported by ESLint. (I think, in the next version of the package I'll provide a short command to launch ESLint more easily, something like `npm run lint`)

**Comment by lydongray** [comment]
Hi. You've certainly knocked that one on the head with your example fiddle. Wrapping the modal in a parent element does create the desired effect I'm looking for. This should work for me so will try it out in my application (I'm using ui-bootstrap modal, so not sure how much control I have over the html rendering). Either way, thank you for you support. I'll close this pull request and related issue for now as this is possible without any code changes. Thanks again.

**Pull request closed without merging**